runtime-light/component/component.h

```diff
@@ -4,78 +4,87 @@
+#include "runtime-light/header.h"
+#include "runtime-light/scheduler/scheduler.h"
+#include "runtime-light/stdlib/fork/fork-context.h"
 #include "runtime-light/stdlib/output-control.h"
 #include "runtime-light/stdlib/rpc/rpc-context.h"
-#include "runtime-light/stdlib/superglobals.h"
-#include "runtime-light/streams/streams.h"
-#include "runtime-light/utils/context.h"
+
+constexpr uint64_t INVALID_PLATFORM_DESCRIPTOR = 0;
+
+// Coroutine scheduler type. Change it here if you want to use another scheduler
+using CoroutineScheduler = SimpleCoroutineScheduler;
+static_assert(CoroutineSchedulerConcept<CoroutineScheduler>);
 
 struct ComponentState {
-  template<typename Key, typename Value>
-  using unordered_map = memory_resource::stl::unordered_map<Key, Value, memory_resource::unsynchronized_pool_resource>;
+  template<typename T>
+  using unordered_set = memory_resource::stl::unordered_set<T, memory_resource::unsynchronized_pool_resource>;
+
   template<typename T>
   using deque = memory_resource::stl::deque<T, memory_resource::unsynchronized_pool_resource>;
-  static constexpr auto INIT_RUNTIME_ALLOCATOR_SIZE = static_cast<size_t>(512U * 1024U); // 512KB
 
-  ComponentState()
+  ComponentState() noexcept
     : runtime_allocator(INIT_RUNTIME_ALLOCATOR_SIZE, 0)
+    , scheduler(runtime_allocator.memory_resource)
+    , fork_component_context(runtime_allocator.memory_resource)
     , php_script_mutable_globals_singleton(runtime_allocator.memory_resource)
-    , opened_streams(unordered_map<uint64_t, StreamRuntimeStatus>::allocator_type{runtime_allocator.memory_resource})
-    , awaiting_coroutines(unordered_map<uint64_t, std::coroutine_handle<>>::allocator_type{runtime_allocator.memory_resource})
-    , timer_callbacks(unordered_map<uint64_t, std::function<void()>>::allocator_type{runtime_allocator.memory_resource})
-    , incoming_pending_queries(deque<uint64_t>::allocator_type{runtime_allocator.memory_resource})
-    , rpc_component_context(runtime_allocator.memory_resource) {}
+    , rpc_component_context(runtime_allocator.memory_resource)
+    , incoming_streams_(deque<uint64_t>::allocator_type{runtime_allocator.memory_resource})
+    , opened_streams_(unordered_set<uint64_t>::allocator_type{runtime_allocator.memory_resource})
+    , pending_updates_(unordered_set<uint64_t>::allocator_type{runtime_allocator.memory_resource}) {}
 
   ~ComponentState() = default;
 
-  bool not_finished() const noexcept {
-    return poll_status != PollStatus::PollFinishedOk && poll_status != PollStatus::PollFinishedError;
-  }
-
-  void resume_if_was_rescheduled();
-
-  bool is_stream_already_being_processed(uint64_t stream_d);
-
-  void resume_if_wait_stream(uint64_t stream_d, StreamStatus status);
+  void init_script_execution() noexcept;
+  void process_platform_updates() noexcept;
 
-  void process_new_input_stream(uint64_t stream_d);
-
-  void init_script_execution();
+  bool stream_updated(uint64_t stream_d) const noexcept {
```

Comment:
I mean also add `inline` keyword. What do you think about `__attribute__ ((always_inline))`? Is it too mush?)

Reply:
No need for that. Functions defined in a class body are implicitly inline: https://en.cppreference.com/w/cpp/language/inline